### PR TITLE
cleanup configuration files

### PR DESCRIPTION
**Pull request by charlierudolph**

### CODE REVIEW

File: src/docker/composebuilder/dependency_builder.go [hunk 1]
```diff
@@ -50,14 +50,13 @@ func (g *LocalDependency) GetDockerConfig() (types.DockerConfig, error) {
 func (g *LocalDependency) GetServiceEnvVariables() map[string]string {
 	result := map[string]string{}
 	result[fmt.Sprintf("%s_HOST", strings.ToUpper(g.name))] = g.name
-	util.Merge(result, g.config.Config.ServiceEnvironment)
```
Comment (charlierudolph) [inline src/docker/composebuilder/dependency_builder.go]: Remove `ServiceEnvironment` as we can use app wide environment variables to achieve the same thing

File: src/types/local_dependency_config.go [hunk 2]
```diff
@@ -1,8 +0,0 @@
-package types
-
-// LocalDependencyConfig represents the configuration of a local dependency
-type LocalDependencyConfig struct {
-	Persist               []string          `yaml:",omitempty"`
-	DependencyEnvironment map[string]string `yaml:"dependency-environment,omitempty"`
-	ServiceEnvironment    map[string]string `yaml:"service-environment,omitempty"`
```
Comment (charlierudolph) [inline src/types/local_dependency_config.go:7]: De-nested these from under Config with the given changes

* Renamed `DependencyEnvironment` to `EnvironmentVariables` for passing only to the dependency
* Added `Secrets` to be consistent with other passing of environment variables
* Removed `ServiceEnvironment` as we can use app wide environment variables to achieve the same thing

File: src/types/remote_dependency.go [hunk 3]
```diff
@@ -9,7 +9,6 @@ import (
 
 // RemoteDependency represents a production dependency
 type RemoteDependency struct {
-	Config         RemoteDependencyConfig `yaml:",omitempty"`
```
Comment (charlierudolph) [inline src/types/remote_dependency.go:12]: Removed this as it is no longer used as we are using TemplateConfig now

File: src/types/service_config.go [hunk 4]
```diff
@@ -21,10 +21,7 @@ const ServiceTypeWorker = "worker"
 // service.yml
 type ServiceConfig struct {
 	Type           string                   `yaml:",omitempty"`
-	Description    string                   `yaml:",omitempty"`
-	Author         string                   `yaml:",omitempty"`
 	DependencyData ServiceDependencyData    `yaml:"dependency-data,omitempty"`
-	Docker         DockerConfig             `yaml:",omitempty"`
```
Comment (charlierudolph) [inline src/types/service_config.go:27]: Removed these fields that are not used

**Review by hugoesthere — APPROVED** [review]

**Pull request merged**